src/structures/Guild.js

```diff
@@ -653,36 +649,29 @@ class Guild extends Base {
    * @param {UserResolvable} user The User to fetch the ban info of
    * @returns {Promise<BanInfo>}
    */
-  fetchBan(user) {
+  async fetchBan(user) {
     const id = this.client.users.resolveID(user);
     if (!id) throw new Error('FETCH_BAN_RESOLVE_ID');
-    return this.client.api
-      .guilds(this.id)
-      .bans(id)
-      .get()
-      .then(ban => ({
-        reason: ban.reason,
-        user: this.client.users.add(ban.user),
-      }));
+    const ban = await this.client.api.guilds(this.id).bans(id).get();
+    return {
+      reason: ban.reason,
+      user: this.client.users.add(ban.user),
+    };
   }
 
   /**
    * Fetches a collection of banned users in this guild.
    * @returns {Promise<Collection<Snowflake, BanInfo>>}
    */
-  fetchBans() {
-    return this.client.api
-      .guilds(this.id)
-      .bans.get()
-      .then(bans =>
-        bans.reduce((collection, ban) => {
-          collection.set(ban.user.id, {
-            reason: ban.reason,
-            user: this.client.users.add(ban.user),
-          });
-          return collection;
-        }, new Collection()),
-      );
+  async fetchBans() {
+    const bans = await this.client.api.guilds(this.id).bans.get();
+    return bans.reduce((collection, ban) => {
```

Comment:
This is the first method I mentioned about using a for loop instead of reduce.

Reply:
May I ask why?

And that goes out of this PR's scope, same goes for the other requests.

Reply:
I mentioned it because it's not really consistent with the other methods of that class the class is using a mix of for-each and reduce and reduce is slower than using for-each.

Reply:
As of Node.js 12, this pattern of reduce (and many others) get translated to nearly the same code as a for-of loop.

Once it's optimised, the performance will be nearly the same, so it really doesn't matter.